blog/_posts/2024-02-26-scala-3.4.0-and-3.3.2-released.md

```diff
@@ -0,0 +1,212 @@
+
+  can now be written more concisely as:
+
+  ```scala
+  val f: [T] => T => String = [T] => x => x.toString
+  ```
+
+- Type inference for functions similar to fold is greatly improved. For example:
+
+  ```scala
+  def partition[T](xs: List[T], pred: T => Boolean) =
+     xs.foldRight((Nil, Nil)): (x, p) =>
+       if pred(x) then (x :: p._1, p._2) else (p._1, x :: p._2)
+
+  ```
+
+  will have its return type correctly inferred as `(List[T], List[T])`. Earlier, it would result in a rather unintuitive type error.
+- The compiler now avoids generating given definitions that loop, removing long-standing footgun of implicit resolution.
+
+### Backend improvements
+
+- Polymorphic lambdas are now implemented using JVM lambdas when possible instead of anonymous classes, which will make them more efficient.
+- JVM Backend parallelization has been ported from Scala 2 to Scala 3. Read more in the discussion under [the original PR](https://github.com/scala/scala/pull/6124).
+
+### Reporting
+
+- If there is an error reading a class file, we now report the version of the classfile and a recommendation to check JDK compatibility:
+
+  ```
+  error while loading String,
+    class file /modules/java.base/java/lang/String.class is broken (version 65.0),
+    please check the JDK compatibility of your Scala version (3.4.0),
+    reading aborted with class java.lang.RuntimeException:
+    foo bar
+  ```
+
+  This will improve the user experience when the class file format changes unexpectedly. This feature will be backported to Scala 3.3.3.
+- Progress reporting of compilation is now visible in IDEs. Metals IDE users will notice that compilation progress is no longer frozen at 0% when using sbt or bloop as the BSP server. IntelliJ will also correctly report progress for BSP and sbt based projects
+
+### Experimental changes
+
+- An experimental `@publicInBinary` annotation can mark definitions that should be treated as a part of binary API. It is useful where some protected or private-in-package definitions are used in the inlined code. When they are marked, it is harder to accidentally break the binary compatibility by doing seemingly harmless refactoring. If the accompanying `-WunstableInlineAccessors` linting option is enabled. There will be a warning about using things not marked as binary API in inlined code. Originaly it was presented in [SIP-52](https://docs.scala-lang.org/sips/binary-api.html).
+- `-experimental` compiler flags will mark all top-level definitions as `@experimental`. This means that the experimental language features and definitions can be used in the project. Note that this does not change the strong guarantees of the stability of the non-experimental code. The experimental features can only be used in an experimental scope (transitively). In 3.5, we plan to allow using this flag also in the stable releases of the compiler.
+
+### Legacy syntax depreacations
+
+Following syntax is now deprecated and will report warnings when used:
```

Comment:
Some of these suggested replacements cannot be used in cross-compiled codebase. Can you advise library maintainers on what `-source` flag will avoid such non-actionable warnings?